src/test/kotlin/kage/ArmorTest.kt

```diff
@@ -0,0 +1,170 @@
+      }
+
+    assertThat(error.message).isEqualTo("trailing data after armored file")
+  }
+
+  @Test
+  fun testTrailingWhitespaceData() {
+    val encryptedInputStr =
+      """-----BEGIN AGE ENCRYPTED FILE-----
+      |YWdlLWVuY3J5cHRpb24ub3JnL3YxCi0+IHNjcnlwdCBEUXh3Qk95OWtmOEdBVFJH
+      |Ukw4aE1RIDE4CnNlY1pRR011ekJpVjZvZUFhakhGZE8rcUtyRWtEN1c4b0hXbzhr
+      |ZG5iekEKLS0tIEYzcXpHb2N4STExV0VEWnZtQUFpRXA4OXBLUWlWTTJnbkhPbEJs
+      |Sy8vNVUKjYeREvMLfFR1ZFUCohjQnSP/d1n4hPuxTeFggWM94q6dhTr6qrvjBMPL
+      |ec4AeigPelkT
+      |-----END AGE ENCRYPTED FILE-----
+      |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               
+      |"""
+        .trimMargin()
+
+    val decryptedOutput = ByteArrayOutputStream()
+
+    val encryptedInput = ByteArrayInputStream(encryptedInputStr.toByteArray())
+
+    val error =
+      assertThrows<ArmorCodingException> {
+        Age.decryptStream(listOf(identity), encryptedInput, decryptedOutput)
+      }
+
+    assertThat(error.message).isEqualTo("too much trailing whitespace")
+  }
+
+  @Test
+  fun testInvalidFirstLine() {
+    val encryptedInputStr =
+      """-----BEGIN AGE ENCRYPTED FILE-----something else
+      |YWdlLWVuY3J5cHRpb24ub3JnL3YxCi0+IHNjcnlwdCBEUXh3Qk95OWtmOEdBVFJH
+      |Ukw4aE1RIDE4CnNlY1pRR011ekJpVjZvZUFhakhGZE8rcUtyRWtEN1c4b0hXbzhr
+      |ZG5iekEKLS0tIEYzcXpHb2N4STExV0VEWnZtQUFpRXA4OXBLUWlWTTJnbkhPbEJs
+      |Sy8vNVUKjYeREvMLfFR1ZFUCohjQnSP/d1n4hPuxTeFggWM94q6dhTr6qrvjBMPL
+      |ec4AeigPelkT
+      |-----END AGE ENCRYPTED FILE-----
+      |"""
+        .trimMargin()
+
+    val decryptedOutput = ByteArrayOutputStream()
+
+    val encryptedInput = ByteArrayInputStream(encryptedInputStr.toByteArray())
+
+    val error =
+      assertThrows<ArmorCodingException> {
+        Age.decryptStream(listOf(identity), encryptedInput, decryptedOutput)
+      }
+
+    assertThat(error.message)
+      .isEqualTo("invalid first line: -----BEGIN AGE ENCRYPTED FILE-----something else")
```

Comment:
```suggestion
    assertThat(error).hasMessageThat()
      .isEqualTo("invalid first line: -----BEGIN AGE ENCRYPTED FILE-----something else")
```

Use [this](https://github.com/google/truth/blob/9db1dddacd63eedcb32a08e24887be130ac7b6fd/core/src/main/java/com/google/common/truth/ThrowableSubject.java#L48) instead.